eds_scikit/plot/event_sequences.py

```diff
@@ -0,0 +1,212 @@
+    family_to_index: Optional[Dict[str, int]] = None,
+    list_person_ids: Optional[List[str]] = None,
+    same_x_axis_scale: Optional[bool] = False,
+    subplot_height: Optional[int] = 200,
+    subplot_width: Optional[int] = 500,
+    point_size: Optional[int] = 400,
+    bar_height: Optional[int] = 20,
+    title: Optional[str] = None,
+    seed: Optional[int] = 0,
+) -> alt.Chart:
+    """
+    Plots individual sequences from an events DataFrame. Each event must be recorded with a start date, a name and a `person_id`.
+    Events can be both one-time (only start date given) or longitudinal (both start and end dates).
+    Events can also be aggregated in families using the `family_col` argument.
+    Finally, events labelling and colors can be manually set by providing a `dim_mapping` dictionary.
+
+    Parameters
+    ----------
+    df_events: pd.DataFrame
+        DataFrame gathering the events information. Must contain at least `person_id`, event, t_start and t_end columns.
+    event_col: Optional[str] = "event"
+        Column name of the events.
+    event_start_datetime_col: Optional[str] = "event_start_datetime"
+        Column name of the event start datetime.
+    event_end_datetime_col: Optional[str] = "event_end_datetime"
+        Column name of the event end datetime.
+    dim_mapping: Optional[Dict[str,Dict[str,Union[tuple(int),str]]]] = None
+        Mapping dictionary to provide plotting details on events. Must be of type :
+        ```python
+            dim_labelling = {
+                "event_1" : {"color":(255,200,150), "label":"Event 1"},
+                "event_2" : {"color":(200,255,150), "label":"Event 2"}
+            }
+        ```
+    index_date_col: Optional[str] = None
+        Column name of the index date to compute relative datetimes for events. For example, it could be the date of inclusion for each patient.
+    family_col: Optional[str] = None
+        Column name of family events. Events of a given family will be plot on the same row.
+    family_to_index: Optional[Dict[str,int]] = None
+        Dictionary mapping event family names to ordering indices.
+    list_person_ids: Optional[List[str]] = None
+        List of person_ids to plot. If None given, only the first three individual sequences will be plot.
+    same_x_axis_scale: Optional[bool] = False
+        Whether to use the same axis scale for all sequences.
+    subplot_height: Optional[int] = 200
+        Height of each plot.
+    subplot_width: Optional[int] = 500
+        Width of each plot.
+    point_size: Optional[int] = 400
+        Size of points for one-time events.
+    bar_height: Optional[int] = 20
+        Height of bars for continuous events.
+    title: Optional[str] = None
+        Chart title.
+    seed: int = 0
+        Seed to randomly draw colors when not provided.
+
+    Returns
+    -------
+    chart: alt.Chart
+        Chart with the plotted individual event sequences.
+    """
+    random.seed(seed)
+
```

Comment:
It might be super helpful to raise errors upfront when some columns are missing in the Dataframe. `check_columns` is probably the easiest way to do so.

https://github.com/aphp/eds-scikit/blob/c15119c91480dc2faed108b966bb0e66b5fbb597/eds_scikit/utils/checks.py#L192

Reply:
You are right thanks, just added !